scripts/globals/extravaganza.lua

```diff
@@ -0,0 +1,73 @@
+-----------------------------------
+-- Alter Ego Extravaganza
+-----------------------------------
+require('scripts/globals/settings')
+require('scripts/globals/items')
+require('scripts/globals/npc_util')
+-----------------------------------
+
+xi = xi or {}
+xi.extravaganza = {}
+
+local enabled = xi.settings.main.ENABLE_TRUST_ALTER_EGO_EXTRAVAGANZA
+
+-----------------------------------------------
+-- Determine Which Ciphers to Grant by NPC Name
+-----------------------------------------------
+
+xi.extravaganza.getShadowEraCiphers = function(npc)
+    local ciphers = {}
+    switch (npc:getName()): caseof
+    {
+        ["Shixo"]   = function (x) ciphers={xi.items.CIPHER_OF_NOILLURIES_ALTER_EGO, xi.items.CIPHER_OF_LEONOYNES_ALTER_EGO} end,
+        ["Shenni"]  = function (x) ciphers={xi.items.CIPHER_OF_ELIVIRAS_ALTER_EGO, xi.items.CIPHER_OF_MAXIMILIANS_ALTER_EGO} end,
+        ["Shuvo"]   = function (x) ciphers={xi.items.CIPHER_OF_LHUS_ALTER_EGO, xi.items.CIPHER_OF_KAYEELS_ALTER_EGO} end,
+    }
+
+    return ciphers
+end
+
+----------------------------------------------------------
+-- Check if Extravaganza is Active, set Ciphers, Launch CS
+----------------------------------------------------------
+
+xi.extravaganza.shadowEraTrigger = function(player, npc, notes)
+    if enabled == 1 or enabled == 3 then -- Available during Summer/NY or when set to have all enabled
+        ciphers = xi.extravaganza.getShadowEraCiphers(npc)
+        player:setLocalVar("ShadowCipher1", ciphers[1])
+        player:setLocalVar("ShadowCipher2", ciphers[2])
+        player:startEvent(7300, 0, notes, 6)
+    else
+        -- TODO: Need Cap of CS with params when event not active
+        -- for now have NPC stare blankly
+    end
+end
+
+------------------------------------------------------
+-- Pull Cipher IDs out of local vars between functions
+-- Choose the right cipher depending on Menu Option
+------------------------------------------------------
+
+xi.extravaganza.shadowEraFinish = function(player, csid, option)
+    local ID = zones[player:getZoneID()]
+    local notes = player:getCurrency("allied_notes")
+    local cipherid = {player:getLocalVar("ShadowCipher1"), player:getLocalVar("ShadowCipher2")}
+
+    if csid == 7300 then -- All 3 Shadow Era Cipher vendors share the same CSID and options
+        if option == 65537 then -- First Cipher
+            choice = 1
```

Comment:
You have to define Choice var as local, probably where you defined all locals above, for consistency